internal/pkg/endpoints/kubernetes_test.go

```diff
@@ -2184,6 +2184,117 @@ func TestPodTargetsPathLabel(t *testing.T) {
 	)
 }
 
+func TestPodTargetsPathAnnotationCorrectQuery(t *testing.T) {
+	t.Parallel()
+
+	assert.ElementsMatch(
+		t,
+		podTargets(&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-pod",
+				Namespace: "test-ns",
+				Annotations: map[string]string{
+					"prometheus.io/path": "/metrics?format=prometheus",
```

Comment:
would be helpful to have 1-2 cases more, or even just this, but with a more complex example?

Reply:
Now that we use `url.Parse`, do you think it's necessary? I think it would be testing `url.Parse` rather than our code.

Reply:
now it is not!